electrumpersonalserver/server/transactionmonitor.py

```diff
@@ -335,13 +335,15 @@ def check_for_reorganizations(self):
                     elements_removed.append(reorgable_tx)
                     self.unconfirmed_txes[txid].extend(scrhashes)
 
-                    #add to history as unconfirmed
-                    txd = self.rpc.call("decoderawtransaction", [tx["hex"]])
-                    new_history_element = self.generate_new_history_element(tx,
-                        txd)
-                    for scrhash in scrhashes:
-                        self.address_history[scrhash]["history"].append(
-                            new_history_element)
+                    #don't add orphans back into history
+                    if tx["category"] != "orphan":
```

Comment:
I'd say replace this with 

> if tx["category"] == "orphan":
>    continue

so that the new lines don't need to be indented as much.

Reply:
`continue`ing here though will then also skip the `for` loop on line 375. Isn't that loop required for deleting the orphan?

Reply:
Yeah it was a bit tricky to figure out what was going on :)